src/components/interaction-designer/resource-editors/ResourceEditor.vue

```diff
@@ -1,92 +1,89 @@
-                <li
-                  v-if="can(['edit-content', 'send-call-to-records'], true) && isFeatureAudioUploadEnabled"
-                  class="nav-item">
-                  <a
-                    href="#"
-                    class="nav-link px-2 py-1"
-                    @click.prevent="triggerRecordViaPhoneFor(languageId)">{{ 'flow-builder.phone-recording' | trans }}</a>
-                </li>
-
-                <li
-                  v-if="isEditable"
-                  class="nav-item">
-                  <a
-                    v-if="isFeatureAudioUploadEnabled"
-                    v-flow-uploader="{
-                      target: route('trees.resumeableAudioUpload'),
-                      token: `${block.uuid}${languageId}`,
-                      accept: 'audio/*'}"
-                    class="nav-link px-2 py-1"
-                    href="#"
-                    @filesSubmitted="handleFilesSubmittedFor(`${block.uuid}:${languageId}`, $event)"
-                    @fileSuccess="handleFileSuccessFor(`${block.uuid}:${languageId}`, languageId, $event)">
-                    {{ 'flow-builder.upload' | trans }}
-                  </a>
-                </li>
-              </ul>
+    <div>
+      <b-tabs>
+        <b-tab
+          v-for="{id: languageId, label: language} in flow.languages"
+          :key="languageId"
+          :title="language || 'flow-builder.unknown-language' | trans"
+          active>
+          <div
+            v-for="(mode, i) in flow.supported_modes"
+            :key="i"
+            class="tab-content-style">
+            <header class="d-flex">
+              <svg-icon :style="{marginTop: '2px'}" :icon="iconsMap.get(mode)" />
+              <h6 class="ml-1">{{ `flow-builder.${mode.toLowerCase()}-content` | trans }}</h6>
+            </header>
+
+            <template v-for="contentType in discoverContentTypesFor(mode)">
+              <!-- todo: it's odd that we pass around a ContentType variant rather than a ContentTypeLangMode variant (aka, mode as external arg) -->
+
+              <resource-variant-text-editor
+                v-if="contentType === SupportedContentType.TEXT"
+                :resource-id="resource.uuid"
+
+                :resource-variant="findOrGenerateStubbedVariantOn(
+                  resource,
+                  {language_id: languageId, content_type: contentType, modes: [mode]})"
+                :mode="mode"
+                :enable-autogen-button="true || enableAutogenButton" />
+
+              <div v-if="contentType === SupportedContentType.AUDIO">
+                <audio-library-selector
+                  :audio-files="availableAudio"
+                  :lang-id="languageId"
+                  :resource-id="resource.uuid"
+                  :selected-audio-file="findOrGenerateStubbedVariantOn(
+                    resource,
+                    {language_id: languageId, content_type: contentType, modes: [mode]}).value" />
+
+                <phone-recorder
```

Comment:
I think the mockup I was looking at had Upload Audio button first, then Record Audio second. Is that what you're seeing as well? Should we swap them?

Reply:
Good catch!  I swapped them